scalafmt-core/shared/src/main/scala/org/scalafmt/internal/Router.scala

```diff
@@ -327,6 +331,16 @@ class Router(formatOps: FormatOps) {
         val singleLineSplit =
           Split(Space, 0)
             .notIf(hasSingleLineComment)
+            // for constructors with empty args lambda
+            // new Foo { () =>
+            //   println("wow")
+            // }
+            .notIf(
+              leftOwner.parent.exists(_.is[Template]) &&
+                leftOwner.parent.exists(_.parent.exists(_.is[NewAnonymous])) &&
+                newlines > 0 &&
```

Comment:
1. i think since we are past the arguments here, the earlier rule should have prohibited a space here and forced a newline after `=>`; without this change, doesn't work?

2. we are not allowed to examine `newlines` for `unfold`. generally, `newlines` should only be used with `sourceIgnored`, as `logical(logical(newlines) || sourceIgnored)`